src/collect/collector.rs

```diff
@@ -146,6 +147,11 @@ impl Collectors {
             self.probes.set_probe_opt(probe::ProbeOption::StackTrace)?;
         }
 
+        // --allow-system-changes requires root.
+        if collect.args()?.allow_system_changes && !Uid::effective().is_root() {
```

Comment:
isn't CAP_NET_ADMIN enough at the moment?

Reply:
> It is probably but I'm thinking if an user request system modifications then it makes sense to just ask for being root. If one wants to run as "unprivileged" (aka CAP_SYS_ADMIN :)) then it's fair to expect them to configure the system as expected.
> 

I agree with this. Also --allow-system-changes is an option that may change its requirements in the future, so the privileges.

In general, I'm more for avoiding inconsistencies (even requesting root like in this case) and documenting (if needed) what it's needed otherwise for non-root tracing.

> The reason behind is the check is done at the core level, which makes things easy and safe but that means it's not easy to have it fine-grained. The other options would be to let modules make their own checks depending on if they are running or not, but that has some implications too (not running in auto-mode when their prerequisites are not satisfied).
> 
> Thoughts?